sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AtomXmlSerializer, HttpOperationResponse } from "@azure/core-http";
+import * as Constants from "../util/constants";
+import { serializeToAtomXmlRequest, deserializeAtomXmlResponse } from "../util/atomXmlHelper";
+
+const requestProperties = ["Name", "Filter", "Action"];
+
+export function buildRuleOptions(name: string, ruleOptions: RuleOptions = {}): InternalRuleOptions {
+  const internalRuleOptions: InternalRuleOptions = Object.assign(ruleOptions, { name: name });
```

Comment:
`Object.assign` mutates the 1st parameter passed in, in this case `ruleOptions`. Since `ruleOptions` is something the customer defines and may decide to re-use across multiple operation calls, we should create a copy of it before mutating.

You can essentially do a shallow-copy by changing your code to the following:
```typescript
const internalRuleOptions: InternalRuleOptions = Object.assign({}, ruleOptions, { name: name });
```
This way, only the empty object gets mutated.

__Note__: This feedback applies anywhere where the code is mutating user-provided objects.

Reply:
Oh, that's a neat copy utility to remember!

Since the function was returning an instance of the object, I assumed it creates a new one.
If it's mutating one of the parameters, I wonder see why does it populate result twice.. :/ the return type and value could have been void I think